### PR TITLE
samples: nrf_desktop: Restore peer on reboot

### PR DESCRIPTION
**Pull request by MarekPieta**
Add restoring last used peer on reboot.

### CODE REVIEW

**Comment by MarekPieta** [comment]
@pdunaj, @Qbicz, could you review ?

File: samples/nrf_desktop/src/modules/ble_bond.c [hunk 1]
```diff
@@ -328,7 +349,66 @@ static void silence_unused(void)
...
+{
+	/* Update peer id - select, then selected (for proper LED update) */
+	struct ble_peer_operation_event *event =
+		new_ble_peer_operation_event();
+
+	event->op = PEER_OPERATION_SELECT;
+	event->arg = cur_peer_id;
+	EVENT_SUBMIT(event);
+
+	struct ble_peer_operation_event *event2 =
```
Comment (Qbicz) [inline samples/nrf_desktop/src/modules/ble_bond.c]: `event_selected` instead of `event2` ?

Comment (MarekPieta) [reply samples/nrf_desktop/src/modules/ble_bond.c]: fixed

File: samples/nrf_desktop/src/modules/ble_bond.c [hunk 2]
```diff
@@ -328,7 +349,66 @@ static void silence_unused(void)
...
+	EVENT_SUBMIT(event_selected);
+
+	return 0;
+}
+
+static int init_settings(void)
+{
+	if (IS_ENABLED(CONFIG_SETTINGS)) {
+		int err;
+		static struct settings_handler sh = {
```
Comment (pdunaj) [inline samples/nrf_desktop/src/modules/ble_bond.c]: static const

File: samples/nrf_desktop/src/modules/ble_bond.c [hunk 3]
```diff
@@ -328,7 +349,66 @@ static void silence_unused(void)
...
+	event_selected->arg = cur_peer_id;
+	EVENT_SUBMIT(event_selected);
+
+	return 0;
+}
+
+static int init_settings(void)
+{
+	if (IS_ENABLED(CONFIG_SETTINGS)) {
+		int err;
```
Comment (pdunaj) [inline samples/nrf_desktop/src/modules/ble_bond.c]: move to line 400

File: samples/nrf_desktop/src/modules/ble_bond.c [hunk 4]
```diff
@@ -380,16 +463,24 @@ static bool event_handler(const struct event_header *eh)
...
 			__ASSERT_NO_MSG(!initialized);
 			initialized = true;
 
-			init();
-
-			module_set_state(MODULE_STATE_READY);
+			if (!init() && !err) {
+				module_set_state(MODULE_STATE_READY);
+			} else {
+				module_set_state(MODULE_STATE_ERROR);
```
Comment (pdunaj) [inline samples/nrf_desktop/src/modules/ble_bond.c]: let's assume error can be set multiple times - we will handle it when needed

File: samples/nrf_desktop/src/modules/ble_bond.c [hunk 5]
```diff
@@ -328,7 +349,66 @@ static void silence_unused(void)
...
+	return 0;
+}
+
+static int commit(void)
+{
+	/* Update peer id - select, then selected (for proper LED update) */
+	struct ble_peer_operation_event *event =
+		new_ble_peer_operation_event();
+
+	event->op = PEER_OPERATION_SELECT;
```
Comment (pdunaj) [inline samples/nrf_desktop/src/modules/ble_bond.c]: it's better to fix leds than sending two events

File: samples/nrf_desktop/src/modules/ble_bond.c [hunk 6]
```diff
@@ -178,6 +181,23 @@ static void select_next(void)
...
 }
 
+static void store_peer_id(u8_t peer_id)
+{
+	if (IS_ENABLED(CONFIG_SETTINGS)) {
+		char key[MAX_KEY_LEN];
+		int err = snprintk(key, sizeof(key), MODULE_NAME "/%s",
+				   PEER_ID_STORAGE_NAME);
+
+		if (err > 0 && err < MAX_KEY_LEN) {
```
Comment (pdunaj) [inline samples/nrf_desktop/src/modules/ble_bond.c]: brackets

File: samples/nrf_desktop/src/modules/ble_bond.c (hunk 2 above)
Comment (MarekPieta) [reply samples/nrf_desktop/src/modules/ble_bond.c]: fixed

File: samples/nrf_desktop/src/modules/ble_bond.c (hunk 3 above)
Comment (MarekPieta) [reply samples/nrf_desktop/src/modules/ble_bond.c]: done

File: samples/nrf_desktop/src/modules/ble_bond.c (hunk 6 above)
Comment (MarekPieta) [reply samples/nrf_desktop/src/modules/ble_bond.c]: done

File: samples/nrf_desktop/src/modules/ble_bond.c (hunk 4 above)
Comment (MarekPieta) [reply samples/nrf_desktop/src/modules/ble_bond.c]: ok, changed

File: samples/nrf_desktop/src/modules/ble_bond.c (hunk 5 above)
Comment (MarekPieta) [reply samples/nrf_desktop/src/modules/ble_bond.c]: ok, so I will make proper changes led_state

**Comment by MarekPieta** [comment]
I made requested changes, led_state was simplified

**Pull request merged**